src/daft-catalog/src/identifier.rs

```diff
@@ -0,0 +1,76 @@
+use std::fmt::Display;
+
+use crate::errors::{Error, Result};
+
+/// An object's namespace (location).
+pub type Namespace = Vec<String>;
+
+/// An object's name.
+pub type Name = String;
+
+/// A reference (path) to some catalog object.
+#[derive(Debug, Clone)]
+pub struct Identifier {
+    namespace: Namespace,
+    name: Name,
+}
+
+impl Identifier {
+    /// Creates an identifier from its namespace and name.
+    pub fn new(namespace: Namespace, name: Name) -> Self {
+        Self { namespace, name }
+    }
+
+    /// Returns the namespace
+    pub fn namespace(&self) -> &Namespace {
+        &self.namespace
+    }
+
+    /// Returns the name
+    pub fn name(&self) -> &Name {
```

Comment:
nit: when working with a borrowed `String`, we'd want to return `&str` instead of `&String`